whois/src/main.rs

```diff
@@ -15,173 +15,456 @@
 // along with polkadot-introspector.  If not, see <http://www.gnu.org/licenses/>.
 
 use clap::{Args, Parser, Subcommand};
+use futures::future;
+use itertools::Itertools;
+use libp2p::{Multiaddr, PeerId};
 use polkadot_introspector_essentials::{
 	api::{
 		api_client::ApiClientMode,
 		executor::{RequestExecutor, RequestExecutorError},
 	},
-	consumer::{EventConsumerInit, EventStream},
 	init,
-	telemetry_feed::{AddedNode, TelemetryFeed},
-	telemetry_subscription::{TelemetryEvent, TelemetrySubscription},
-	types::{AccountId32, SessionKeys},
+	types::{AccountId32, H256},
 	utils,
 };
-use polkadot_introspector_priority_channel::Receiver;
-use std::str::FromStr;
+use serde::{Deserialize, Serialize};
+use serde_binary::binary_stream::Endian;
+use ss58_registry::Ss58AddressFormat;
+use std::{
+	collections::{HashMap, HashSet},
+	fs::{self, File},
+	io::Write,
+};
+use subp2p_explorer::util::{crypto::sr25519, p2p::get_peer_id};
+use subp2p_explorer_cli::commands::authorities::PeerDetails;
 
 #[derive(Clone, Debug, Parser)]
-#[clap(author, version, about = "Simple telemetry feed")]
-struct TelemetryOptions {
+#[clap(author, version, about = "Simple command to query polkadot validator identities")]
+struct WhoIsOptions {
 	#[clap(subcommand)]
 	command: WhoisCommand,
 	/// Web-Socket URLs of a relay chain node.
 	#[clap(long)]
 	pub ws: String,
-	/// Web-Socket URL of a telemetry backend
-	#[clap(long)]
-	pub feed: String,
-	/// Name of a chain to connect
-	#[clap(long)]
-	pub chain: Option<String>,
 	#[clap(flatten)]
 	pub verbose: init::VerbosityOptions,
 	#[clap(flatten)]
 	pub retry: utils::RetryOptions,
+	/// The session index used for computing the validator indices.
+	#[clap(long)]
+	pub session_index: u32,
+	/// An optional block hash to fetch the queued keys at, otherwise we use the queued keys at the current block.
+	#[clap(long)]
+	pub queued_keys_at_block: Option<H256>,
```

Comment:
Could you help me to understand when we need it?

Reply:
In case you work with sessions from the past and those keys don't exist in queued keys at the current session, so you would need to use the keys from the past. Unfortunately, this runtime API does not receive a session index, so you need to provide a specific block hash. 

Normally, you wouldn't need to pass anything in there since the current session keys are good enough.